distributed/core.py

```diff
@@ -543,7 +543,8 @@ async def _close_on_failure(exc: Exception) -> None:
             except Exception as exc:
                 await _close_on_failure(exc)
                 raise RuntimeError(f"{type(self).__name__} failed to start.") from exc
-            self.status = Status.running
+            if self.status == Status.init:
+                self.status = Status.running
```

Comment:
the problem in https://github.com/dask/distributed/issues/7699 was that `nanny.close()` gets called during `nanny.start()` so close sets `self.status = Status.closing` but this line changed it back to `self.status = Status.running` which causes the `_on_worker_exit` to attempt a restart

Reply:
Nice catch!